l10n_ch_fds_upload_sepa/wizard/fds_inherit_sepa_wizard.py

```diff
@@ -0,0 +1,319 @@
+        This Class inherit from l10n_ch_sepa.wiz_pain_001.
+    '''
+    _inherit = 'wizard.pain001'
+
+    fds_account = fields.Many2one(
+        comodel_name='fds.postfinance.account',
+        string='FDS account',
+        help='select one FDS account or create a FDS account'
+    )
+    fds_account_directory = fields.Many2one(
+        comodel_name='fds.postfinance.files.directory',
+        string='FDS files directory',
+        help='select one upload directory or config your upload directory'
+    )
+    attachment_id = fields.Many2one(
+        comodel_name='ir.attachment',
+        string='File attachment',
+        readonly=True,
+        help='[info] keep one recored of the model attachment'
+    )
+    filename = fields.Char(
+        string='Attachment filename',
+        related='attachment_id.name',
+        readonly=True,
+        help='The name of the attachment file'
+    )
+    payment_order_id = fields.Many2one(
+        comodel_name='payment.order',
+        string='Payment order',
+        readonly=True,
+        help='payment order id'
+    )
+    state = fields.Selection(
+        selection=[('default', 'Default'),
+                   ('finish', 'finish'),
+                   ('upload', 'upload'),
+                   ('confirm', 'confirm')],
+        readonly=True,
+        default='default',
+        help='[Info] keep state of the wizard'
+    )
+
+    ##################################
+    #         Button action          #
+    ##################################
+    @api.multi
+    def generateFile_button(self):
+        ''' create a pain_001 file into wizard and add it as an attachment
+            Called by pressing generate button.
+
+            :returns action: configuration for the next wizard's view
+        '''
+        self.ensure_one()
+        self.create_pain_001()
+        attachment_obj = self.env['ir.attachment']
+        payment_order_id = self.env.context.get('active_id')
+        file = attachment_obj.search([['res_id', '=', payment_order_id]])[0]
+        self.write({'attachment_id': file.id})
+        self.write({'payment_order_id': payment_order_id})
+        self.write({'state': 'finish'})
+        return self._do_populate_tasks()
+
+    @api.multi
+    def upload_generateFile_button(self):
+        ''' change the view to the wizard or directly upload the file if
+            only one FDS account and only one upload directory selected.
+            Called by pressing upload button.
+
+            :returns action: configuration for the next wizard's view
+        '''
+        self.ensure_one()
+        self._state_upload_on()
+
+        # check if only one fds account
+        existing_account_ids = self.fds_account.search([]).ids
+        if len(existing_account_ids) != 1:
+            return self._do_populate_tasks()
+
+        self.fds_account = existing_account_ids[0]
+
+        # check if default upload directory exist
+        if not self.fds_account.upload_sepa_directory:

```

Comment:
Correct me if I'm wrong, but the search is done on `self.fds_account`

Thus you assign your result to `self.fds_account`

And finally you check that `self.fds_account` has a `upload_sepa_directory`

Sounds to me you can write:

    existing_account_ids = self.fds_account.search([('upload_sepa_directory', '=', 'True']).ids
    if len(existing_account_ids) != 1:
        return self._do_populate_tasks()
    self.fds_account = existing_account_ids[0]


Unless you search on 'fds.postfinance.account' model can return you multiple results and you want to call _do_poplate_tasks in that case.